### PR TITLE
[FIX] pos_viva_wallet: add webhook fallback

### PR DESCRIPTION
**Pull request by MaxTheWhale**
The Viva Wallet integration relies on Odoo
being called by Viva Wallet via a webhook.
There have been many issues of this webhook
never being called, and therefore payments
getting stuck in Odoo.

In this commit, we add a polling fallback.
Every 5 seconds we query the Viva Wallet
session, and if we get a response (success/failure) then we update the payment state in the PoS.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by vlst-odoo** [comment]
hello @MaxTheWhale ,

The idea here is simply that viva wallet sometimes fails to send the request on the webhook or are there also problems with the odoo code ?

**Comment by MaxTheWhale** [comment]
> hello @MaxTheWhale ,
> 
> The idea here is simply that viva wallet sometimes fails to send the request on the webhook or are there also problems with the odoo code ?

That's my understanding, that they (recently) have been failing to send the webhook request
@Ysoroko can confirm the details

File: addons/pos_viva_wallet/models/pos_payment_method.py [hunk 1]
```diff
@@ -153,6 +152,13 @@ def viva_wallet_send_payment_cancel(self, data):
         endpoint = f"sessions/{session_id}?cashRegisterId={cash_register_id}"
         return self._call_viva_wallet(endpoint, 'delete')
 
+    def viva_wallet_get_payment_status(self, session_id):
+        if not self.env.user.has_group('point_of_sale.group_pos_user'):
+            raise AccessError(_("Only 'group_pos_user' are allowed to fetch token from Viva Wallet"))
```
Comment (Ysoroko) [inline addons/pos_viva_wallet/models/pos_payment_method.py:157]: adapt the error message according to the method

**Review by Ysoroko — APPROVED** [review]
Works perfectly, just adapt the access error message

**Comment by Ysoroko** [comment]
@robodoo delegate+

**Comment by MaxTheWhale** [comment]
@robodoo r+

**Pull request closed without merging**